internal/httpctx/context.go

```diff
@@ -26,3 +28,15 @@ func (v *Value[T]) Set(r *http.Request, val T) *http.Request {
 	*r = *h
 	return h
 }
+
+var selfLoopingValue = NewValue[bool](ctx.SelfLooping)
```

Comment:
Since these are new concerns, please just move the new code into `looping.go` and `looping_test.go`.
Should avoid some potential cherry picks.